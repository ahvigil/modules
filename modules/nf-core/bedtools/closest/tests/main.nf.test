nextflow_process {
    name "Test Process BEDTOOLS_CLOSEST"
    script "../main.nf"
    process "BEDTOOLS_CLOSEST"

    tag "modules"
    tag "modules_nfcore"
    tag "bedtools"
    tag "bedtools/closest"

    test("test_bedtools_closest") {
        when {
            process {
                """                
                input[0] = [
                    [ id:'test' ], // meta map
                    file(params.test_data['homo_sapiens']['genome']['genome_bed'], checkIfExists: true),
                    [
                        file(params.test_data['homo_sapiens']['genome']['genome_multi_interval_bed'], checkIfExists: true),
                        file(params.test_data['homo_sapiens']['genome']['genome_21_multi_interval_bed'], checkIfExists: true)
                    ]
                ]
                input[1] = []
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("test_bedtools_closest_fasta_fai") {
        when {
            process {
                """                
                input[0] = [
                    [ id:'test' ], // meta map
                    file(params.test_data['homo_sapiens']['genome']['genome_bed'], checkIfExists: true),
                    file(params.test_data['homo_sapiens']['genome']['genome_multi_interval_bed'], checkIfExists: true)
                ]
                input[1] = file(params.test_data['homo_sapiens']['genome']['genome_fasta_fai'], checkIfExists: true)
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("test_bedtools_closest_vcf") {
        when {
            process {
                """                
                input[0] = [
                    [ id:'test' ], // meta map
                    file(params.test_data['homo_sapiens']['genome']['genome_bed'], checkIfExists: true),
                    [
                        file(params.test_data['homo_sapiens']['genome']['dbsnp_146_hg38_vcf_gz'], checkIfExists: true),
                        file(params.test_data['homo_sapiens']['genome']['gnomad_r2_1_1_vcf_gz'], checkIfExists: true)
                    ]
                ]
                input[1] = []
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
