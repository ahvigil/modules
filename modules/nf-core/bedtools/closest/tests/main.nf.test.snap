{
    "homo_sapiens - vcf": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bed:md5,85a5cee50322ab15f384df7826164d64"
                    ]
                ],
                "1": [
                    "versions.yml:md5,21b18c873c7ca8f5166982532fd82266"
                ],
                "output": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bed:md5,85a5cee50322ab15f384df7826164d64"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,21b18c873c7ca8f5166982532fd82266"
                ]
            }
        ],
        "timestamp": "2023-11-28T15:02:05.619655"
    },
    "homo_sapiens - fai": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bed:md5,347f192c141a34dfc4bb7c13fffd44bb"
                    ]
                ],
                "1": [
                    "versions.yml:md5,21b18c873c7ca8f5166982532fd82266"
                ],
                "output": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bed:md5,347f192c141a34dfc4bb7c13fffd44bb"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,21b18c873c7ca8f5166982532fd82266"
                ]
            }
        ],
        "timestamp": "2023-11-28T15:01:59.924973"
    },
    "homo_sapiens": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bed:md5,de47864c0ae53891d09252eb63cd7269"
                    ]
                ],
                "1": [
                    "versions.yml:md5,21b18c873c7ca8f5166982532fd82266"
                ],
                "output": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bed:md5,de47864c0ae53891d09252eb63cd7269"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,21b18c873c7ca8f5166982532fd82266"
                ]
            }
        ],
        "timestamp": "2023-11-28T15:01:54.666124"
    }
}